### PR TITLE
Support scoop package also 

### PR DESCRIPTION
**Pull request by cderv**
This PR is just an update to the README because everything is done in the scoop bucket I created already for rstudio-daily installation: https://github.com/cderv/r-bucket

You can see the README there and the PR https://github.com/cderv/r-bucket/pull/1

As it is pretty simple to create a manifest for support a package, I added the 3 binaries. `TinyTex-1.zip`, `TinyTex-0.zip` and `TinyTex.zip` where not explicit names so I used `TinyTex` (for Rmd support - recommanded one I guess), `TinyTex-full` and `TinyTex-min`. Notice I kept the upper T in the name. 

* **Should we use `tinytex` as for choco ?**
* **Should we offer only one package instead ?**

For the install step I did the same as in your choco script
* tlmgr path add (without admin right, so in user mode)
* creating a shim only for `tlmgr`

**Is this correct ?** 
I was note sure what was the purpose of 
https://github.com/yihui/tinytex-releases/blob/617697a37f6bbc3cb591fd271ac91f9460718646/choco/tools/chocolateyinstall.ps1#L18-L22
but I supposed it was choco-only related.

IMO This works as expected. 

And also the **r-bucket** is automatically updated thanks to a GHA workflow. So as soon as a release is done here, the manifest in the bucket will be updated. 

@yihui I let you look into it, and discuss here. I will make the change in the bucket accordingly. 

We could also decide to make `TinyTex-releases` as bucket - but there would only be TinyTex in it. 
We could also make a PR into https://github.com/ScoopInstaller/Main or https://github.com/lukesampson/scoop-extras which the 2 default bucket you got setup when installing scoop.

### CODE REVIEW

**Comment by naveen521kk** [comment]
> Should we use tinytex as for choco ?
Idk about that. But choco can't have id in caps. Also, I don't think it is necessary to create a package for the first one `TinyTeX-0.*` because it contains only infra-only and nobody requires it (AFAIK).

https://github.com/yihui/tinytex-releases/blob/617697a37f6bbc3cb591fd271ac91f9460718646/choco/tools/chocolateyinstall.ps1#L18-L22
This is something totally related to chocolatey. It creates a shim for every `exe` files which should not be done as we add to path and that's why that code generates a `.ignore` file to make it ignore those. https://chocolatey.org/docs/features-shim

**Comment by cderv** [comment]
> Also, I don't think it is necessary to create a package for the first one TinyTeX-0.* because it contains only infra-only and nobody requires it (AFAIK).

I was thinking of now windows R user that want a quick install and also for CI workflow where you would want to control installed Tex packages to keep them at the minimum. But I guess install those packages would take time whereas installing the binaries would be quicker, right ?

**Comment by naveen521kk** [comment]
> But I guess install those packages would take time whereas installing the binaries would be quicker, right ?

Exactly. If they want infra-only scheme they get one quickly (around 3 min in my PC) from `install-tl.zip`. So others can also do that.

File: README.md [hunk 1]
```diff
@@ -32,6 +32,57 @@ To uninstall TinyTeX, use the command:
 choco uninstall tinytex
 ```
 
-## Scoop package
+## Scoop packages
 
-Coming...
+Scoop is another package manager for Windows. You need to install [_scoop_](https://scoop-docs.now.sh/docs/getting-started/Quick-Start.html) first to use it from powershell.
+
+Apps for _scoop_ are available through "buckets". For now, TinyTex binary packages are not available in the default **main** or **extras** buckets that comes with a new installation of _scoop_, but through the scoop bucket **r-bucket** at https://github.com/cderv/r-bucket/ with other R-related tools.
```
Comment (yihui) [inline README.md:39]: `TinyTex` -> `TinyTeX` ([I have to say this every time](https://yihui.org/en/2017/12/typing-names/) 😂 )

File: README.md [hunk 2]
```diff
@@ -32,6 +32,57 @@ To uninstall TinyTeX, use the command:
...
+
+* Minimal installation (infra-only) of `TinyTeX-0.zip`
+```powershell
+scoop install TinyTex-min
+```
+
+* Full installation (more packages) of `TinyTeX-1.zip`
+```powershell
+scoop install TinyTex-min
+```
```
Comment (yihui) [inline README.md:61]: Let's only provide `TinyTeX-1.zip` for now, and see what users think. If necessary, we could replace it with `TinyTeX.zip` in the future.

File: README.md [hunk 3]
```diff
@@ -32,6 +32,57 @@ To uninstall TinyTeX, use the command:
...
+Starting download ...
+Download: (OK):download completed.
+Checking hash of TinyTeX-0.zip ... ok.
+Extracting TinyTeX-0.zip ... done.
+Running pre-install script...
+--> Another TinyTex already found. Cancelling current installation...
+Uninstalling 'TinyTex-min' (2020.09).
+'TinyTex-min' was uninstalled.
+Exception: You already have Tinytex installed. Run scoop uninstall Tinytex if you want to use TinyTex-min.
+```
```
Comment (yihui) [inline README.md:79]: Let's keep it brief:

```suggestion
For more information, please see https://github.com/cderv/r-bucket/.
```

**Review by yihui — APPROVED** [review]
> * Should we use `tinytex` as for choco ?

@cderv Yes, let's just use the name lowercase `tinytex` for all non-LaTeX packages, and reserve `TinyTeX` for the LaTeX distribution.

> * Should we offer only one package instead ?

I tend to offer `TinyTeX-1` for both chocolatey and scoop (and perhaps Homebrew in the future).

`TinyTeX-0` is for those who don't want my selection of packages: https://github.com/yihui/tinytex/blob/master/tools/pkgs-custom.txt They can use `TinyTeX-0` as the base, and install any LaTeX packages that they want. It could also be used for CI workflows.

> We could also decide to make `TinyTex-releases` as bucket - but there would only be TinyTex in it.

No need to make this repo a bucket. As you said, it doesn't make much sense to have a bucket that only contains a single package.

> We could also make a PR into https://github.com/ScoopInstaller/Main or https://github.com/lukesampson/scoop-extras which the 2 default bucket you got setup when installing scoop.

I don't care much if TinyTeX is in the main bucket (because I don't use it myself), although it would be good if it could be added there. Apparently I'll not be the best person to maintain this scoop package. If you think you could maintain it, please feel free to send them a PR. Thanks!

**Review by naveen521kk — APPROVED** [review]
Do the changes @yihui asked and it is good to go.

File: README.md (hunk 2 above)
Comment (cderv) [reply README.md:50]: The doc was incorrect here :) 

`tinytex` scoop package will provide TinyTeX-1.zip

File: README.md (hunk 1 above)
Comment (cderv) [reply README.md]: oups 

I made the change!

File: README.md [hunk 4]
```diff
@@ -32,7 +32,7 @@ To uninstall TinyTeX, use the command:
 choco uninstall tinytex
 ```
 
-## Scoop packages
+## Scoop package
```
Comment (naveen521kk) [inline README.md:35]: Just a nitpick
```suggestion
## Scoop Package
```

Comment (cderv) [reply README.md:35]: I did the same as for chocolatey package
https://github.com/yihui/tinytex-releases/blob/master/README.md#L19

No uppercase P 😄

Comment (naveen521kk) [reply README.md:35]: Do for both. 😉

Comment (cderv) [reply README.md:35]: It is just a convention right ? 
As we used Sentence case for the first header, I prefer not to use Title Case just for these two other header. I prefer to leave that out for now.

**Comment by cderv** [comment]
@naveen521kk, at the end you are the reviewer of this PR as a collaborator. I am not one yet for now, so I can't merge the PR :) 

I think it is ready - Regarding Sentence case vs Title Case, it is a convention to apply on all title. You can pick one 😉 

@yihui I took into account your comment. I only show scoop `tinytex` 📦 here but kept `tinytex-full` and `tinytex-min` in the **r-bucket**

What we may want to document or deal with is : What should happen with update ? Chocolatey and Scoop are designed to make updating easy (`choco upgrade` or `scoop update`) - When a new release will be made, the updated package will be seen but I am not sure one should update monthly right ? What happens with the Tex package installed in that case ?
In scoop I can block the update so that `scoop status` won't  show an updated `tinytex`

**Comment by naveen521kk** [comment]
> Chocolatey and Scoop are designed to make updating easy (choco upgrade or scoop update) - When a new release will be made, the updated package will be seen but I am not sure one should update monthly right ? 

Users would need to update once a year because once a year TeXLive updates it. But we can't keep LaTeX packages outdated. So our part is to update it once a month(or maybe soon as done in other Debian packages by for TexLive). Meanwhile, users can use `tlmgr` to update individual packages.

> What happens with the Tex package installed and not part of the binary in that case ?
Those will get deleted. In chocolatey, `ToolsDir`(Where TinyTeX is installed) is wiped in an upgrade. So they would need to install it again.

**Review by naveen521kk — APPROVED** [review]
LGTM!

**Comment by cderv** [comment]
> Those will get deleted. In chocolatey, ToolsDir(Where TinyTeX is installed) is wiped in an upgrade. So they would need to install it again.

That is what I thought. I don't know enough TeX Live but is there a way to have a library external to the installation folder ? Or is it a bad idea ? With scoop, a folder of the installation can be made persistent. Useful for configuration files for example. 
I have also wondered this for CI worflows - should the all installation folder be cached or can we cache only the packages ? 

@yihui we may need to update the document in https://yihui.org/tinytex/ right ?

**Comment by yihui** [comment]
@cderv You brought up a really good question. When we update the `tinytex` package in scroop or choco, we certainly don't want users to lose the extra LaTeX packages that they installed by themselves via `tlmgr`. This situation is like R. I think users would need a personal TEXMF tree (like an R library) to store these extra packages, which won't be overridden if they update `tinytex` itself. I'll investigate further. Thanks!

**Comment by cderv** [comment]
Glad I am not completly out of scope ! 

This question emerged also when I was trying to use caching in GHA for the setup-tinytex action (https://github.com/r-lib/actions/tree/master/setup-tinytex) - it would be interesting to have a specific path to cache (like the R library) ! 

I let you investigate and I'll help implement if needed. Thanks !

**Comment by naveen521kk** [comment]
> it would be interesting to have a specific path to cache (like the R library indeed) 

In windows, everything is inside that one folder. I think. I had set up [here](https://github.com/ManimCommunity/manim/blob/master/.github/workflows/ci.yml#L84) and all work properly. SO caching that folder would do.

**Comment by cderv** [comment]
Yes but you are caching the whole distribution installation - and that works but it feels "different" for other tools with package (R, Python, node, ...). But the issue here is mainly with updating from scoop or chocolatey that would remove user installed packages - IMO it is not  a desired behavior if we have a solution.

**Comment by naveen521kk** [comment]
The most relevant doc available releted to the issue is https://www.tug.org/texlive/doc/tlmgr.html#USER-MODE possibly @yihui can you update the TinyTeX doc asking people to use that?

**Comment by yihui** [comment]
Yes, I'll document it. They only need a personal TEXMF tree if they install TinyTeX via tools like `choco` or `scoop`. This is not necessary if they use the `.bat` or `.sh` scripts to install.

**Comment by naveen521kk** [comment]
Is it something like environment variable? Then we can very well do it in chocolatey, or wait I will have a custom update script in chocolatey and use that for upgrading. Is it simply calling `tlmgr self update` or something similar. What is my plan is we can set an environment variable say `TINYTEXDIR` where it points to install bin directory. If it is there then we can use `tlmgr` to update if not then use the normal install script. Can be easily implemented in chocolatey. I don't know about scoop.

**Comment by yihui** [comment]
Not an environment variable, but a TeX Live setting (with `tlmgr conf texmf TEXMFHOME`): https://www.tug.org/texlive/doc/tlmgr.html#conf

**Comment by naveen521kk** [comment]
I will make chocolatey handle it. What can possibly be done is to say chocolatey not to wipe the directory and just update it using the commands
```
tlmgr update  --all
tlmgr update --self
```
Look's like it itself updates the infrastructure. https://www.tug.org/texlive/doc/tlmgr.html#update-option...-pkg

**Comment by yihui** [comment]
If there is a way for choco not to wipe the previous installation but run some commands before updating to a newer version of a package, that'll be great, and we wouldn't need a TEXMFHOME setting. Basically what you need to do is save the list of installed LaTeX packages, and reinstall them after updating the choco package. Here is how it could be done on *nix: https://github.com/yihui/tinytex/blob/31122bdefa4078c0a70089cf9c11fdf4afc3b985/tools/install-bin-unix.sh#L5-L9 You might be able to figure out how to do it on Windows.

**Comment by naveen521kk** [comment]
Ok, I will have some time to work tomorrow and I will do.

**Pull request merged**